native_client/deepspeech.cc

```diff
@@ -485,22 +485,23 @@ Metadata* ModelState::decode_metadata(vector<float>& logits)
 {
   vector<Output> out = decode_raw(logits);
 
-  Metadata* metadata = (Metadata*)malloc(sizeof (Metadata));
+  Metadata* metadata = new Metadata;
   metadata->num_items = out[0].tokens.size();
-  metadata->items = (MetadataItem*)malloc(sizeof(MetadataItem) * metadata->num_items);
+  metadata->items = new MetadataItem[metadata->num_items];
 
   // Loop through each character
   for (int i = 0; i < out[0].tokens.size(); ++i) {
-    MetadataItem item;
-    item.character = (char*)alphabet->StringFromLabel(out[0].tokens[i]).c_str(); 
-    item.timestep = out[0].timesteps[i]; 
-    item.start_time = static_cast<float>(out[0].timesteps[i] * AUDIO_WIN_STEP);
+    MetadataItem *item = new MetadataItem;
```

Comment:
@reuben Correct me if I'm wrong, but I guess this one should also be `std::unique_ptr<>` ?